### PR TITLE
add and upgrade notes entry for renamed jmxfetch metrics

### PR DESCRIPTION
**Pull request by arbll**

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -24,6 +24,13 @@ The Datadog Agent now automatically look for the container short image name to s
 The source is especially important as it triggers the **automatic configuration of your platform with integration pipeline and facets**.
 The Datadog Agent autodiscovery can still be used to override the default source and service with pod annotations or container labels.
 
+Upgrade Notes
+-------------
+
+- The agent now requires a cluster agent version 1.0+ to establish
+  a valid connection
+- JMX garbage collection metrics ``jvm.gc.cms.count`` and ``jvm.gc.parnew.time`` were renamed in 6.6 to be more meaningful. To ensure backward compatibility the change was reverted in this release and the new names put behind a config option. If you started relying on this new names please enable the ``new_gc_metrics`` option in your jmx configurations. An example can be found `here <https://github.com/DataDog/datadog-agent/blob/1aee233a18dedbb8af86da0ce1f2e305206aacf8/cmd/agent/dist/conf.d/jmx.d/conf.yaml.example#L8-L13>`_. This flag will be enabled by default in a future major release.
```
Comment (olivielpeau) [inline CHANGELOG.rst]: `these new names`

Comment (olivielpeau) [reply CHANGELOG.rst]: also, I think it's worth mentioning the new metric names here as well (even if they're listed in the `conf.yaml.example` file). Makes this note easier to understand at a glance IMO.

**Review by olivielpeau — DISMISSED** [review]
Thanks for adding this 🙇  Left a couple of comments, feel free to merge once addressed

Note: will need to be cherry-picked to `6.8.x` and updated on the github release as well.

Note 2: great idea to put the `upgrade notes` right after the `prelude`, could you make that change effective in the reno config? (in a separate PR :) )

File: releasenotes/config.yaml [hunk 2]
```diff
@@ -12,6 +12,11 @@ template: |
           #
           # Each section note must be formatted as reStructuredText.
           ---
+          upgrade:
```
Comment (olivielpeau) [inline releasenotes/config.yaml]: this is cool, but I guess what actually determines the order of the sections in the generated changelog is what's defined under the `sections` key below no?

**Review by olivielpeau — DISMISSED** [review]
Another comment, but feel free to merge this and address later

File: releasenotes/config.yaml (hunk 2 above)
Comment (arbll) [reply releasenotes/config.yaml]: 🤦‍♂️

**Pull request merged**